src/test/java/org/folio/circulation/resources/RequestQueueResourceTest.java

```diff
@@ -0,0 +1,315 @@
+      new ReorderQueueBuilder()
+        .addReorderRequest(recallRequest.getId().toString(), 1)
+        .addReorderRequest(inFulfillmentRequest.getId().toString(), 2)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody(),
+      is("Requests can not be displaced from position 1 when fulfillment begun."));
+  }
+
+  @Test
+  public void validationErrorWhenNonExistingRequestProvidedInReorderedQueue() throws Exception {
+    loansFixture.checkOutByBarcode(item, usersFixture.rebecca());
+
+    IndividualResource firstRecallRequest = recallRequest(steve);
+    IndividualResource secondRecallRequest = recallRequest(jessica);
+
+    Response response = requestQueueFixture.attemptReorderQueue(
+      item.getId().toString(),
+      new ReorderQueueBuilder()
+        .addReorderRequest(secondRecallRequest.getId().toString(), 1)
+        .addReorderRequest(firstRecallRequest.getId().toString(), 2)
+        .addReorderRequest(UUID.randomUUID().toString(), 3)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody(),
+      is("There is inconsistency between provided reordered queue and item queue."));
+  }
+
+  @Test
+  public void validationErrorWhenNotAllRequestsProvidedInReorderedQueue() throws Exception {
+    loansFixture.checkOutByBarcode(item, usersFixture.rebecca());
+
+    holdRequest(steve);
+    IndividualResource secondRecallRequest = recallRequest(jessica);
+
+    Response response = requestQueueFixture.attemptReorderQueue(
+      item.getId().toString(),
+      new ReorderQueueBuilder()
+        .addReorderRequest(secondRecallRequest.getId().toString(), 1)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody(),
+      is("There is inconsistency between provided reordered queue and item queue."));
+  }
+
+  @Test
+  public void canNotSaveRequestsWithDuplicatedPositions() throws Exception {
+    loansFixture.checkOutByBarcode(item, usersFixture.rebecca());
+
+    IndividualResource holdRequest = holdRequest(steve);
+    IndividualResource recallRequest = recallRequest(jessica);
+
+    Response response = requestQueueFixture.attemptReorderQueue(
+      item.getId().toString(),
+      new ReorderQueueBuilder()
+        .addReorderRequest(recallRequest.getId().toString(), 1)
+        .addReorderRequest(holdRequest.getId().toString(), 1)
+        .create()
+    );
+
+    assertThat(response.getStatusCode(), is(500));
+    assertThat(response.getBody(),
+      is("Some requests have not been updated. Fetch the queue and try again."));
+  }
+
+  @Test
+  @Parameters({
+    "1, 2, 3, 4",
+    "1, 2, 4, 3",
+    "2, 1, 3, 4",
+    "2, 1, 4, 3",
+    "4, 3, 2, 1",
+    "4, 3, 1, 2",
+    "3, 4, 2, 1",
+    "3, 4, 1, 2",
+    "3, 4, 1, 50",
+    "40, 2, 1, 3",
+  })
+  public void shouldReorderQueueSuccessfully(int firstPosition, int secondPosition,
```

Comment:
What do the examples with 40 and 50 in them mean?

Reply:
Asked Cate in Jira, waiting for her response.

Reply:
Removed this case, we must not allow such update.